pephubclient/pephubclient.py

```diff
@@ -1,177 +1,338 @@
+        :param jwt_data: jwt string
+        :return: Authorization dict
+        """
         if jwt_data:
-            return {"pephub_session": jwt_data}
+            return {"Authorization": jwt_data}
         else:
             return {}
 
-    def _load_pep_project(self, pep_project: str) -> peppy.Project:
-        FilesManager.save_pep_project(
-            pep_project, self.registry_path, filename=self.DEFAULT_PROJECT_FILENAME
-        )
-        project = Project(self.DEFAULT_PROJECT_FILENAME)
-        FilesManager.delete_file_if_exists(self.DEFAULT_PROJECT_FILENAME)
-        return project
-
-    def _build_request_url(self, variables: dict) -> str:
-        endpoint = (
-            self.registry_path.namespace
-            + "/"
-            + self.registry_path.item
-            + "/"
-            + PEPHubClient.CONVERT_ENDPOINT
-        )
-        if variables:
-            variables_string = PEPHubClient._parse_variables(variables)
+    def _build_pull_request_url(self, query_param: dict = None) -> str:
+        """
+        Build request for getting projects form pephub
+        :param query_param: dict of parameters used in query string
+        :return: url string
+        """
+        if not query_param:
+            query_param = {}
+        query_param["tag"] = self.registry_path.tag
+
+        endpoint = self.registry_path.namespace + "/" + self.registry_path.item
+        if query_param:
+            variables_string = PEPHubClient._parse_query_param(query_param)
             endpoint += variables_string
         return PEPHUB_PEP_API_BASE_URL + endpoint
 
     @staticmethod
-    def _parse_variables(pep_variables: dict) -> str:
+    def _build_push_request_url(namespace: str) -> str:
+        """
+        Build project uplaod request used in pephub
+        :param namespace: namespace where project will be uploaded
+        :return: url string
+        """
+        return PEPHUB_PUSH_URL.format(namespace=namespace)
+
+    @staticmethod
+    def _parse_query_param(pep_variables: dict) -> str:
         """
         Grab all the variables passed by user (if any) and parse them to match the format specified
         by PEPhub API for query parameters.
-
-        Returns:
-            PEPHubClient variables transformed into string in correct format.
+        :param pep_variables: dict of query parameters
+        :return: PEPHubClient variables transformed into string in correct format.
         """
         parsed_variables = []
 
         for variable_name, variable_value in pep_variables.items():
             parsed_variables.append(f"{variable_name}={variable_value}")
-
         return "?" + "&".join(parsed_variables)
+
+    @staticmethod
+    def _handle_pephub_response(pephub_response: requests.Response):
+        """
+        Check pephub response
+        """
+        decoded_response = PEPHubClient.decode_response(pephub_response)
+
+        if pephub_response.status_code != ResponseStatusCodes.OK_200:
```

Comment:
Can the previous statement alter the value of `pephub_response.status_code`? If not, this conditional should be moved to the start of the function, to potentially avoid executing the previous statement.